File: src/LeafSimilarTree.java

class Solution {

    public class TreeNode {
      int val;
      TreeNode left;
      TreeNode right;
      TreeNode(int x) { val = x; }
    }
    
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        return generateLeafStr(root1).equals(generateLeafStr(root2));
    }
    
    private String generateLeafStr(TreeNode root){
        if(root == null)
            return "";
        
        if(root.left == null && root.right == null)
            return Integer.toString(root.val);
        
        String left = generateLeafStr(root.left); 
        String right = generateLeafStr(root.right);
        return left+right;
    }
}

File: src/JavaScript/LeafSimilarTree.js

function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
var leafSimilar = function(root1, root2) {
    return generateStr(root1) === generateStr(root2);
};

const generateStr = (root) => {
    if(root === null)
        return "";
    
    if(root.left === null && root.right === null)
        return ""+root.val;
    
    let left = generateStr(root.left);
    let right = generateStr(root.right);
    return left+right;
}